app/request.go

```diff
@@ -60,14 +85,108 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
 			Flags: []cli.Flag{
 				&cli.StringFlag{
 					Name:     "request-id",
-					Usage:    "The request-id of the asynchronous request",
+					Usage:    "The request-id of the request",
+					Aliases:  []string{"r"},
+					Required: true,
+				},
+			},
+			Action: func(ctx *cli.Context) error {
+				res, err := c.getRequestStatus(ctx, ctx.String("request-id"))
+				if err != nil {
+					return err
+				}
+				return PrintProto(res)
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait for the request complete",
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "request-id",
+					Usage:    "the request-id of the request",
 					Aliases:  []string{"r"},
 					Required: true,
 				},
 			},
 			Action: func(ctx *cli.Context) error {
-				return c.getRequestStatus(ctx.String("request-id"))
+				return c.waitOnRequest(ctx, "", ctx.String("request-id"))
 			},
 		}},
 	}}, nil
 }
+
+func (c *RequestClient) waitOnRequest(ctx *cli.Context, operation string, requestID string) error {
+
+	ticker := time.NewTicker(time.Millisecond)
+	timer := time.NewTimer(ctx.Duration(RequestTimeoutFlagName))
+
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
+
+	var status *request.RequestStatus
+	defer func() {
+		if status != nil {
+			PrintProto(status)
+		}
+	}()
+loop:
+	for {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timed out waiting for request to complete, namespace=%s, requestID=%s, timeout=%s",
+				ctx.String(NamespaceFlagName),
+				requestID,
+				ctx.Duration(RequestTimeoutFlagName),
+			)
+		case <-ticker.C:
+			var err error
+			status, err = c.getRequestStatus(ctx, requestID)
+			if err != nil {
+				return err
+			}
+			switch status.State {
+			case request.STATE_FULFILLED:
+				break loop
+			case request.STATE_FAILED:
+				fmt.Fprintf(writer, "operation failed \n")
+				return fmt.Errorf("request failed: %s", status.FailureReason)
+			case request.STATE_CANCELLED:
+				fmt.Fprintf(writer, "operation cancelled\n")
+				return fmt.Errorf("request was cancelled: %s", status.FailureReason)
+			}
+			if operation != "" {
+				fmt.Fprintf(writer, "waiting for %s operation (requestId='%s') to finish, current state: %s\n",
+					operation, requestID, request.State_name[int32(status.State)])
+			} else {
+				fmt.Fprintf(writer, "waiting for request with id='%s' to finish, current state: %s\n",
+					requestID, request.State_name[int32(status.State)])
+			}
+			ticker.Reset(time.Second * time.Duration(status.CheckDuration.Seconds))
```

Comment:
If `status.CheckDuration.Seconds` is 0 `ticker.Reset` [will panic](https://pkg.go.dev/time#Ticker.Reset). Should we set a minimum value to ensure that this never happens and ensure that the ticker fires less often than 1ms?

Reply:
Good catch, fixed.